.github/workflows/end2end.yaml

```diff
@@ -187,9 +187,25 @@ jobs:
           password: ${{ secrets.ARTIFACTS_PASSWORD }}
           source: ${{ env.ROOT_DOCUMENTATION_BUILDDIR }}
 
+
+  mongo-patches-check:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: run patches
+        run: make -C solution-base/mongodb patch
+      - name: Ensure mongodb charts are patched
+        run: |
+          git diff --exit-code || { echo "Error: Changes detected after running 'make fetch-mongodb-sharded'. Please ensure step is completed."; exit 1; }
+        if: always()
```

Comment:
why `always` ?
- if checkout fails, no point doing anything more
- `make patch` should not fail unless there are changes to apply, and these fail to apply --> if it fails, not point doing git diff